### PR TITLE
feat: handle public $cObj and add set function

### PR DESCRIPTION
**Pull request by helsner**
Refs #2623

### CODE REVIEW

**Comment by helsner** [comment]
@TomasVotruba Hi Tomas, i would like to take the chance to get to know you! 😄 
Maybe you got a hint for me as well as its basic rector

I need to change the visibility of the property from `public` to `protected` do i do this by remove and readd or is there another way?

File: src/Rector/v11/v4/ProvideCObjViaMethodRector.php [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        if ($cObjProperty->isPublic()) {
+            // TODO: add handling to make it protected
+        }
+
+        $classMethod = $node->getMethod('setContentObjectRenderer');
+        if ($classMethod instanceof ClassMethod) {
+            return null;
+        }
+
+        $this->addSetContentObjectRendererMethod($node);
```
Comment (TomasVotruba) [inline src/Rector/v11/v4/ProvideCObjViaMethodRector.php:65]: Here the modified class node should be returned, as you're adding an item to array.
That's my rough guess :)

File: src/Rector/v11/v4/ProvideCObjViaMethodRector.php [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    /**
+     * @param Class_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($this->isObjectType($node, new ObjectType('TYPO3\CMS\Frontend\Plugin\AbstractPlugin'))
+            || $this->isObjectType($node, new ObjectType('TYPO3\CMS\Extbase\Mvc\Controller\ActionController'))
+        ) {
+            return null;
+        }
```
Comment (TomasVotruba) [inline src/Rector/v11/v4/ProvideCObjViaMethodRector.php:49]: So if the class is type A or B, it should be skipped?
Should be inverted? :)

**Comment by sabbelasichon** [comment]
@helsner Thanks a lot, well done. Let´s ship it to the public.

**Pull request merged**